### PR TITLE
generate batch report

### PR DESCRIPTION
**Pull request by crewtool**
# Description
This PR adds option to generate raport of max batch which fits GPU

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/train/configs/baselines/gpt/granularity/medium.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-parent: research/conditional/train/configs/baselines/gpt/base.yaml
-md5_parent_hash: 1cc2f52ca625d4d24f327c4874413d0c
+parent: research/conditional/train/configs/baselines/gpt/medium.yaml
+md5_parent_hash: 335962f2f67ef04105ef58db9f3da0d7
```
Comment (Simontwice) [inline research/conditional/train/configs/baselines/gpt/granularity/medium.yaml:2]: is this related to the PR? doesnt seem so

File: research/conditional/utils/argparse.py [hunk 2]
```diff
@@ -101,6 +101,9 @@ def introduce_parser_arguments(
     parser.add_argument("--git_branch", type=str, default=None)
     parser.add_argument("--decoding_interval", type=int, default=0)
 
+    parser.add_argument("--model_fits_filename", type=str, default=None)
+    parser.add_argument("--model_fits_params", type=str, default=None)
```
Comment (Simontwice) [inline research/conditional/utils/argparse.py:105]: rename it to gpu_usage_report_params
```suggestion
    parser.add_argument("--gpu_usage_report_params", type=str, default=None)
```

File: research/conditional/utils/argparse.py [hunk 3]
```diff
@@ -101,6 +101,9 @@ def introduce_parser_arguments(
     parser.add_argument("--git_branch", type=str, default=None)
     parser.add_argument("--decoding_interval", type=int, default=0)
 
+    parser.add_argument("--model_fits_filename", type=str, default=None)
```
Comment (Simontwice) [inline research/conditional/utils/argparse.py:104]:
```suggestion
    parser.add_argument("--gpu_usage_report_path", type=str, default=None)
```

Comment (Simontwice) [reply research/conditional/utils/argparse.py:104]: remember to refactor

File: research/conditional/utils/argparse.py (hunk 2 above)
Comment (Simontwice) [reply research/conditional/utils/argparse.py:105]: remember to refactor

File: lizrd/generate_raport_model_fits.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Generate raport model fits")
+    parser.add_argument(
+        "--gradient_checkpointing",
+        action="store_true",
+        help="Is table for gradient checkpointing",
+    )
+    parser.add_argument("--input_file", default="results.json", type=str)
```
Comment (Simontwice) [inline lizrd/generate_raport_model_fits.py:63]:
```suggestion
    parser.add_argument("--input_file", default="gpu_usage_report.json", type=str)
```

File: lizrd/generate_raport_model_fits.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    parser.add_argument("--group_sizes", default="2,4,8,16,32,64,128,256", type=str)
```
Comment (Simontwice) [inline lizrd/generate_raport_model_fits.py:64]: no defaults

File: research/conditional/train/cc_train.py [hunk 6]
```diff
@@ -220,6 +226,8 @@ def main(
         min_eval_group_size=args.min_eval_group_size,
         max_eval_group_size=args.max_eval_group_size,
         steps_until_start_temperature_learn=args.steps_until_start_temperature_learn,
+        zipped_model_fits_params=zipped_model_fits_params,
```
Comment (Simontwice) [inline research/conditional/train/cc_train.py:229]: How about we unpack this to a list already? so the arguments are as self-explanatory as possible.

Comment (Simontwice) [reply research/conditional/train/cc_train.py:229]: unpack before***

File: lizrd/generate_raport_model_fits.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import yaml
+import argparse
+import json
+import pandas as pd
```
Comment (Simontwice) [inline lizrd/generate_raport_model_fits.py:4]: I don't know how to mark it, but a file rename would be good here: maybe `generate_gpu_usage_report`?

File: research/conditional/train/configs/baselines/gpt/mot/generate_max_batch_size_report/base.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    save_weights_interval: 0
+    logging_interval_heavy: 100000000000
+
+    use_dummy_dataset: true
+
+    init_type: "kaiming_uniform"
+    init_scale: 1.0
+
+    model_fits_filename: "results_base.json"
+    model_fits_params: "dmodel,dff,n_blocks,n_att_heads,group_size,gradient_checkpointing,batch_size"
```
Comment (Simontwice) [inline research/conditional/train/configs/baselines/gpt/mot/generate_max_batch_size_report/base.yaml:34]: I do not understand this. Shouldn't the things we fit over be the ones we put in grid args? (^ delimiter above)

File: lizrd/generate_raport_model_fits.py (hunk 5 above)
Comment (crewtool) [reply lizrd/generate_raport_model_fits.py:64]: IMO we should keep defaults in scripts which are not a trainer

File: research/conditional/train/configs/baselines/gpt/mot/generate_max_batch_size_report/base.yaml (hunk 8 above)
Comment (crewtool) [reply research/conditional/train/configs/baselines/gpt/mot/generate_max_batch_size_report/base.yaml]: I needed a way to distinguish different model sizes, therefore I have used "dmodel,dff,n_blocks,n_att_heads". Other args are the same as we grid over

File: lizrd/scripts/generate_gpu_max_batch_size_report.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+from lizrd.support.misc import merge_dicts
+
+
+def main(
+    gradient_checkpointing: bool,
+    group_sizes: [int],
+    models: [str],
+    report_files: [str],
+    output_file,
+):
```
Comment (Simontwice) [inline lizrd/scripts/generate_gpu_max_batch_size_report.py:15]: I really need a docstring here.
What are the inputs? outputs?
what fields does the yaml config require?

Comment (crewtool) [reply lizrd/scripts/generate_gpu_max_batch_size_report.py]: outdated

**Review by Simontwice — APPROVED** [review]
v nice code quality! LGTM
